src/core/runner.ts

```diff
@@ -277,142 +270,133 @@ export default class Runner {
         (skipStep && !step.only) ||
         (isOnlyExists && !step.only)
       ) {
-        data.status = 'skipped';
+        step.status = 'skipped';
       } else {
-        data = await this.runStep(step, context, options);
+        data = await this.#runStep(step, options);
         /**
          * skip next steps if the previous step returns error
          */
-        if (data.error && !step.soft) skipStep = true;
+        if (step.error && !step.soft) skipStep = true;
       }
-      this.#reporter?.onStepEnd?.(journey, step, {
-        start,
-        end: monotonicTimeInSeconds(),
-        ...data,
-      });
-      if (options.pauseOnError && data.error) {
+      step.duration = monotonicTimeInSeconds() - step._startTime;
+      this.#reporter?.onStepEnd?.(journey, step, data);
+      if (options.pauseOnError && step.error) {
         await new Promise(r => process.stdin.on('data', r));
       }
       results.push(data);
     }
     return results;
   }
 
-  registerJourney(journey: Journey, context: JourneyContext) {
-    this.#currentJourney = journey;
-    const timestamp = getTimestamp();
-    const { params } = context;
+  async #startJourney(journey: Journey, options: RunOptions) {
+    journey._startTime = monotonicTimeInSeconds();
+    this.#driver = await Gatherer.setupDriver(options);
+    await Gatherer.beginRecording(this.#driver, options);
+    /**
+     * For each journey we create the screenshots folder for
+     * caching all screenshots and clear them at end of each journey
+     */
+    await mkdir(this.#screenshotPath, { recursive: true });
+    const params = options.params
     this.#reporter?.onJourneyStart?.(journey, {
-      timestamp,
+      timestamp: getTimestamp(),
       params,
     });
     /**
      * Exeucute the journey callback which registers the steps for current journey
      */
-    journey.callback({ ...context.driver, params });
+    journey.cb({ ...this.#driver, params, info: this });
   }
 
-  async endJourney(
-    journey,
-    result: JourneyContext & JourneyResult,
+  async #endJourney(
+    journey: Journey,
+    result: JourneyResult,
     options: RunOptions
   ) {
-    const end = monotonicTimeInSeconds();
-    const { pluginManager, start, status, error, browserDelay } = result;
-    const pluginOutput = await pluginManager.output();
+    // Enhance the journey results
+    const pOutput = await Gatherer.pluginManager.output();
+    const bConsole = filterBrowserMessages(pOutput.browserconsole, journey.status);
     await this.#reporter?.onJourneyEnd?.(journey, {
-      status,
-      error,
-      start,
-      end,
-      browserDelay,
+      browserDelay: this.#browserDelay,
       timestamp: getTimestamp(),
       options,
-      ...pluginOutput,
-      browserconsole: filterBrowserMessages(
-        pluginOutput.browserconsole,
-        status
-      ),
+      networkinfo: pOutput.networkinfo,
+      browserconsole: bConsole,
```

Comment:
status is removed as value but is still typed in JourneyEndResult.

Reply:
Its because of the `Partial<Journey>`, Have to clean this up and use the `RunnerInfo`. Will do that when we do the cleanup for Multi Step API tests.